src/contexts/AuthContext/index.tsx

```
@@ -0,0 +1,78 @@
+
+/**
+ * 4. 認証機能を提供するプロバイダーを実装
+ * => アプリケーション全体で認証状態を管理・共有できるようにするためのコンポーネント
+ */
+export const AuthContextProvider = ({
+  context,
+  authUser,
+  children
+}: React.PropsWithChildren<AuthContextProviderProps>) => {
+  // ユーザー情報を取得するSWRフック
+  const { data, error, mutate } = useSWR<User>(`${context.apiRootUrl.replace(/\/$/g, '')}/users/me`);
```

ESLint

[error] 'error' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)